jdi-light-bootstrap-tests/src/test/java/io/github/epam/bootstrap/tests/common/TooltipTests.java

```diff
@@ -0,0 +1,163 @@
+                {tooltipOnRightButton, textOnRightTooltip},
+                {tooltipWithHTML, textOnHTMLTooltip},
+                {wrapperForDisabledButton, textOnDisabledButton}
+        };
+    }
+
+    @DataProvider(name = "Buttons")
+    public static Object[][] buttonsText() {
+        return new Object[][]{
+                {tooltipOnTopButton, topButtonText},
+                {tooltipOnBottomButton, bottomButtonText},
+                {tooltipOnLeftButton, leftButtonText},
+                {tooltipOnRightButton, rightButtonText},
+                {tooltipOnBottomButton, bottomButtonText},
+                {tooltipWithHTML, htmlButtonText}
+        };
+    }
+
+    @Test(dataProvider = "Tooltips")
+    public void getTooltipTextTests(Tooltip tooltip, String tooltipText) {
+        assertEquals(tooltip.getTooltipText(), tooltipText);
+    }
+
+    @Test(dataProvider = "Tooltips")
+    public void tooltipIsInvisibleTests(Tooltip tooltip, String __) {
+        assertFalse(tooltip.isVisible());
+    }
+
+    @Test(dataProvider = "Tooltips")
+    public void tooltipIsVisibleByHoverTests(Tooltip tooltip, String __) {
+        tooltip.core().hover();
+        assertTrue(tooltip.isVisible());
+    }
+
+    @Test(dataProvider = "Tooltips")
+    public void tooltipIsVisibleByFocusTests(Tooltip tooltip, String __) {
+        tooltip.core().focus();
+        assertTrue(tooltip.isVisible());
+    }
+
+    @Test(dataProvider = "Tooltips")
+    public void tooltipIsVisibleByClickTests(Tooltip tooltip, String __) {
+        tooltip.click();
+        assertTrue(tooltip.isVisible());
+    }
+
+    @Test(dataProvider = "TooltipsPlacements")
+    public void tooltipPlacementTests(Tooltip tooltip, String placement) {
+        assertEquals(tooltip.getTooltipPlacement(), placement);
+    }
+
+    @Test
+    public void tooltipWithHtmlTest(){
+        assertTrue(tooltipWithHTML.isTooltipWithHTML());
+    }
+
+    @Test(dataProvider = "TooltipsPlacements")
+    public void tooltipWithoutHtmlTest(Tooltip tooltip, String __){
+        assertFalse(tooltip.isTooltipWithHTML());
+    }
+
+    @Test
+    public void tooltipOnDisabledButtonIsInvisibleByHoverTest() {
+        tooltipOnDisabledButton.core().hover();
+        assertFalse(tooltipOnDisabledButton.isVisible());
+    }
+
+    @Test
+    public void tooltipOnDisabledButtonIsInvisibleByFocusTest() {
+        tooltipOnBottomButton.core().focus();
+        assertFalse(tooltipOnDisabledButton.isVisible());
+    }
+
+    @Test
+    public void isValidationTestForDisabledButton() {
+        tooltipOnDisabledButton.is()
+                .displayed()
+                .text(disabledButtonText)
+                .core()
+                .cssClass("btn btn-info btn-block")
+                .tag(is("button"));
+        try {
+            tooltipOnDisabledButton.click();
+            fail("Disabled button should not work, but work");
+        } catch (Exception ex) {
+            assertThat(safeException(ex),
```

Comment:
could you please introduce me a bit about that code:
```
assertThat(safeException(ex),
           containsString("Can't perform click. Element is disabled"));
```
it can be useful for another people)

Reply:
To write code in a similar style, I've rewritten this method implementation from disableButtonTest, which is located here: https://github.com/jdi-testing/jdi-light/blob/bootstrap/jdi-light-bootstrap-tests/src/test/java/io/github/epam/bootstrap/tests/common/ButtonTests.java